alteryx/woodwork
# v0.1.0

**v0.1.0 March 22, 2021**
- Enhancements
    - Implement Schema and Accessor API (#497)
    - Add Schema class that holds typing info (#499)
    - Add WoodworkTableAccessor class that performs type inference and stores Schema (#514)
    - Allow initializing Accessor schema with a valid Schema object (#522)
    - Add ability to read in a csv and create a DataFrame with an initialized Woodwork Schema (#534)
    - Add ability to call pandas methods from Accessor (#538, #589)
    - Add helpers for checking if a column is one of Boolean, Datetime, numeric, or categorical (#553)
    - Add ability to load demo retail dataset with a Woodwork Accessor (#556)
    - Add ``select`` to WoodworkTableAccessor (#548)
    - Add ``mutual_information`` to WoodworkTableAccessor (#571)
    - Add WoodworkColumnAccessor class (#562)
    - Add semantic tag update methods to column accessor (#573)
    - Add ``describe`` and ``describe_dict`` to WoodworkTableAccessor (#579)
    - Add ``init_series`` util function for initializing a series with dtype change (#581)
    - Add ``set_logical_type`` method to WoodworkColumnAccessor (#590)
    - Add semantic tag update methods to table schema (#591)
    - Add warning if additional parameters are passed along with schema (#593)
    - Better warning when accessing column properties before init (#596)
    - Update column accessor to work with LatLong columns (#598)
    - Add ``set_index`` to WoodworkTableAccessor (#603)
    - Implement ``loc`` and ``iloc`` for WoodworkColumnAccessor (#613)
    - Add ``set_time_index`` to WoodworkTableAccessor (#612)
    - Implement ``loc`` and ``iloc`` for WoodworkTableAccessor (#618)
    - Allow updating logical types with ``set_types`` and make relevant DataFrame changes (#619)
    - Allow serialization of WoodworkColumnAccessor to csv, pickle, and parquet (#624)
    - Add DaskColumnAccessor (#625)
    - Allow deserialization from csv, pickle, and parquet to Woodwork table (#626)
    - Add ``value_counts`` to WoodworkTableAccessor (#632)
    - Add KoalasColumnAccessor (#634)
    - Add ``pop`` to WoodworkTableAccessor (#636)
    - Add ``drop`` to WoodworkTableAccessor (#640)
    - Add ``rename`` to WoodworkTableAccessor (#646)
    - Add DaskTableAccessor (#648)
    - Add Schema properties to WoodworkTableAccessor (#651)
    - Add KoalasTableAccessor (#652)
    - Adds ``__getitem__`` to WoodworkTableAccessor (#633)
    - Update Koalas min version and add support for more new pandas dtypes with Koalas (#678)
    - Adds ``__setitem__`` to WoodworkTableAccessor (#669)
- Fixes
    - Create new Schema object when performing pandas operation on Accessors (#595)
    - Fix bug in ``_reset_semantic_tags`` causing columns to share same semantic tags set (#666)
    - Maintain column order in DataFrame and Woodwork repr (#677)
- Changes
    - Move mutual information logic to statistics utils file (#584)
    - Bump min Koalas version to 1.4.0 (#638)
    - Preserve pandas underlying index when not creating a Woodwork index (#664)
    - Restrict Koalas version to ``<1.7.0`` due to breaking changes (#674)
    - Clean up dtype usage across Woodwork (#682)
    - Improve error when calling accessor properties or methods before init (#683)
    - Remove dtype from Schema dictionary (#685)
    - Add ``include_index`` param and allow unique columns in Accessor mutual information (#699)
    - Include DataFrame equality and ``use_standard_tags`` in WoodworkTableAccessor equality check (#700)
    - Remove ``DataTable`` and ``DataColumn`` classes to migrate towards the accessor approach (#713)
    - Change ``sample_series`` dtype to not need conversion and remove ``convert_series`` util (#720)
    - Rename Accessor methods since ``DataTable`` has been removed (#723)
- Documentation Changes
    - Update README.md and Get Started guide to use accessor (#655, #717)
    - Update Understanding Types and Tags guide to use accessor (#657)
    - Update docstrings and API Reference page (#660)
    - Update statistical insights guide to use accessor (#693)
    - Update Customizing Type Inference guide to use accessor (#696)
    - Update Dask and Koalas guide to use accessor (#701)
    - Update index notebook and install guide to use accessor (#715)
    - Add section to documentation about schema validity (#729)
    - Update README.md and Get Started guide to use ``pd.read_csv`` (#730)
    - Make small fixes to documentation formatting (#731)
- Testing Changes
    - Add tests to Accessor/Schema that weren't previously covered (#712, #716)
    - Update release branch name in notes update check (#719)
